cfme/common/provider.py

```diff
@@ -97,6 +97,7 @@ class BaseProvider(Taggable, Updateable, Navigatable, BaseEntity):
     def __attrs_post_init__(self):
         # attr.ib(convert=prepare_endpoints) doesn't work correctly. this is workaround
         self.endpoints = prepare_endpoints(self.endpoints)
+        self.rest_api_entity = self.appliance.rest_api.collections.providers.get(name=self.name)
```

Comment:
**REQUIRED:** This breaks ```get_crud``` if the provider hasn't been added to the appliance. We should do some checking here, remember we can instantiate without it being on the appliance. I think perhaps it would be better to have a sync method. This will perhaps look for it and sync certain values over time, but perhaps to start with it just syncs the id?